### PR TITLE
Introduce WordPress 5.8 compatiblity

### PR DESCRIPTION
**Pull request by olafleur**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -145,7 +145,7 @@ commands:
             sudo apt-get update && sudo apt-get install -y subversion default-mysql-client
             sudo -E docker-php-ext-install mysqli
             mkdir -p /tmp/wordpress
-            ./vendor/bin/wp core download --path=/tmp/wordpress
+            ./vendor/bin/wp core download --path=/tmp/wordpress --version=nightly
```
Comment (ehamwey) [inline .circleci/config.yml:148]: We should ensure this doesn't make it into master, although it should stay for the moment for testing purposes.

Comment (olafleur) [reply .circleci/config.yml:148]: Yes. Good point. I'll keep your comment as a reminder to remove this line before merging ;)

File: package.json [hunk 2]
```diff
@@ -69,9 +69,14 @@
 		"zh_TW": "繁體中文"
 	},
 	"dependencies": {
+		"@babel/plugin-transform-regenerator": "^7.14.5",
+		"@babel/plugin-transform-runtime": "^7.14.5",
+		"@babel/runtime": "^7.14.6",
```
Comment (ehamwey) [inline package.json:74]: another note - according to https://core.trac.wordpress.org/ticket/52941#comment:24 it appears babel/polyfill will be returning to wordpress rather than the regenerator runtime. Either way this shouldn't be necessary once the change gets into the next nightly build.

File: src/blocks/icon/svgs.js [hunk 3]
```diff
@@ -1596,7 +1596,7 @@ let icons = bundledIconsEnabled ? {
 			__( 'sitemap', 'coblocks' ),
 		],
 		icon: <SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><Path d="M0 0h24v24H0z" fill="none" /><Path d="M22 11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3z" /></SVG>,
-		icon_outlined: <SVG xmlns="http://www.w3.org/2000/svg" enable-background="new 0 0 24 24" viewBox="0 0 24 24"><Rect fill="none" height="24" width="24" /><Path d="M22,11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3H22z M7,9H4V5h3V9z M17,15h3v4h-3V15z M17,5h3v4h-3V5z" /></SVG>,
+		icon_outlined: <SVG xmlns="http://www.w3.org/2000/svg" enableBackground="new 0 0 24 24" viewBox="0 0 24 24"><Rect fill="none" height="24" width="24" /><Path d="M22,11V3h-7v3H9V3H2v8h7V8h2v10h4v3h7v-8h-7v3h-2V8h2v3H22z M7,9H4V5h3V9z M17,15h3v4h-3V15z M17,5h3v4h-3V5z" /></SVG>,
```
Comment (ehamwey) [inline src/blocks/icon/svgs.js:1599]: Should these be a part of @godaddy-wordpress/coblocks-icons?

File: package.json (hunk 2 above)
Comment (ehamwey) [reply package.json:74]: This also applies to: https://github.com/godaddy-wordpress/coblocks/pull/1923/files#diff-9fd21b09a0144355cfd001e3952767df08b2e04241994470aa1d627f797b171e and https://github.com/godaddy-wordpress/coblocks/pull/1923/files#diff-379120de71991cfea15f15a1225d1ba83a7a50e55a356c831f5f5e0be305ce0b

File: .circleci/config.yml (hunk 1 above)
Comment (ehamwey) [reply .circleci/config.yml:148]: This also applies to https://github.com/godaddy-wordpress/coblocks/pull/1923/files#diff-9fd21b09a0144355cfd001e3952767df08b2e04241994470aa1d627f797b171e unless we want wp-env to stay on nightly builds.

Comment (olafleur) [reply .circleci/config.yml:148]: Yeah, for the .wp-env.json config file it's a little bit different, as it's only for the local execution. Maybe it shouldn't be there, but it matters less if it's merged or not, because it doesn't impact the CI.

File: src/blocks/icon/svgs.js (hunk 3 above)
Comment (AnthonyLedesma) [reply src/blocks/icon/svgs.js:1599]: They probably should be. That change is out of the scope of this PR but should be a story on the board for sure.

File: src/blocks/form/test/form.cypress.js [hunk 4]
```diff
@@ -6,6 +6,8 @@
  */
 import * as helpers from '../../../../.dev/tests/cypress/helpers';
 
+// TODO : there seems to be a real bug here, as the success message is not shown and the form is still there
+
```
Comment (AnthonyLedesma) [inline src/blocks/form/test/form.cypress.js:10]: Let's follow up on this. I could not replicate this.

File: src/blocks/hero/test/hero.cypress.js [hunk 5]
```diff
@@ -110,13 +110,5 @@ describe( 'Test CoBlocks Hero Block', function() {
...
 
 		cy.get( '.media-toolbar-primary > .button' ).click();
-
-		cy.get( '.media-replace-flow button' ).click();
-
-		cy.get( '.components-popover__content' ).should( 'be.visible' );
-
-		cy.get( '.block-editor-media-replace-flow__media-upload-menu .components-menu-item__button' ).contains( 'Open Media Library' ).click();
-
-		cy.get( '.components-popover__content' ).should( 'not.be.visible' );
```
Comment (AnthonyLedesma) [inline src/blocks/hero/test/hero.cypress.js:120]: Why was this removed?

File: src/extensions/lightbox-controls/test/lightbox-controls.cypress.js [hunk 6]
```diff
@@ -45,7 +45,7 @@ describe( 'Test CoBlocks Lightbox Controls extension', function() {
 			helpers.addBlockToPost( 'core/image', true );
 
 			helpers.upload.imageToBlock( 'core/image' );
-
+			cy.wait( 500 );
```
Comment (AnthonyLedesma) [inline src/extensions/lightbox-controls/test/lightbox-controls.cypress.js:48]: What was this? I think the need for this was fixed by commit https://app.circleci.com/pipelines/github/godaddy-wordpress/coblocks/6222/workflows/84aa207a-2dbd-4e71-8f55-57b614e6e7b7

File: src/extensions/media-text-styles/test/media-text-styles.cypress.js [hunk 7]
```diff
@@ -15,7 +15,10 @@ describe( 'Test CoBlocks Media Text styles extension', function() {
 		const selectBlock = () => cy.get( `img[src*="${ imageBase }"]` ).click();
 
 		cy.get( `img[src*="${ imageBase }"]` );
-		cy.get( '.wp-block-media-text__content p' ).first().type( 'Content' );
+
+		// Cypress cannot type content in a p, it seems
+		//cy.get( '.wp-block-media-text__content p' ).first().type( 'Content', { force: true } );
+
```
Comment (AnthonyLedesma) [inline src/extensions/media-text-styles/test/media-text-styles.cypress.js:21]: I think we need to try to `.focus()` on the p tag before we type.

File: .babelrc [hunk 8]
```diff
@@ -3,6 +3,12 @@
...
   ],
   "plugins": [
-	"@babel/plugin-proposal-class-properties"
+	"@babel/plugin-proposal-class-properties",
+  [
+    "@babel/plugin-transform-runtime",
+    {
+      "corejs": 2,
+    }
+  ]
```
Comment (ehamwey) [inline .babelrc:12]:
```suggestion
 	"@babel/plugin-proposal-class-properties"
```
With the updates in 5.8-beta3 these are no longer necessary.

File: package.json [hunk 9]
```diff
@@ -69,9 +69,14 @@
...
+		"@babel/runtime-corejs2": "^7.14.6",
```
Comment (ehamwey) [inline package.json:75]:
```suggestion
```
These can also be removed as of 5.8-beta3

**Review by ehamwey — CHANGES_REQUESTED** [review]

File: src/blocks/form/test/form.cypress.js (hunk 4 above)
Comment (olafleur) [reply src/blocks/form/test/form.cypress.js:10]: Oh! Will try that first thing tomorrow morning when I'll be back!

File: src/blocks/hero/test/hero.cypress.js (hunk 5 above)
Comment (olafleur) [reply src/blocks/hero/test/hero.cypress.js:120]: Because when executing it, the Replace flow button was not there, so I infered it had been removed from the Hero block. But in fact, maybe it was because it's a bug... Do you know if it is supposed to be there?

File: src/blocks/dynamic-separator/test/dynamic-separator.cypress.js [hunk 10]
```diff
@@ -127,6 +127,7 @@ describe( 'Test CoBlocks Dynamic Seperator Block', function() {
 	/**
 	 * Test dynamic separator custom class
 	 */
+	// pointer-events: none does not allow to click in addCustomBlockClass function
```
Comment (olafleur) [inline src/blocks/dynamic-separator/test/dynamic-separator.cypress.js:130]: Feel free to remove the comment if you are not able to reproduce the failing test.

File: src/blocks/icon/test/icon.cypress.js [hunk 11]
```diff
@@ -166,6 +166,7 @@ describe( 'Test CoBlocks Icon Block', function() {
 		helpers.editPage();
 	} );
 
+	// pointer-events: none does not allow us to click
```
Comment (olafleur) [inline src/blocks/icon/test/icon.cypress.js:169]: Feel free to remove the comment if you are not able to reproduce the failing test.

File: src/blocks/share/test/share.cypress.js [hunk 12]
```diff
@@ -88,6 +89,7 @@ describe( 'Test CoBlocks Share Block', function() {
 	/**
 	 * Test the coblocks share block button size.
 	 */
+	// Weird CSS behaviour : real width does not match CSS width
```
Comment (olafleur) [inline src/blocks/share/test/share.cypress.js:92]: Feel free to remove the comment if you are not able to reproduce the failing test.

File: src/blocks/share/test/share.cypress.js [hunk 13]
```diff
@@ -210,6 +212,7 @@ describe( 'Test CoBlocks Share Block', function() {
 	/**
 	 * Test the coblocks share block custom classes & top/bottom spacing.
 	 */
+	// `pointer-events: none` prevents user mouse interaction (herited from a parent)
```
Comment (olafleur) [inline src/blocks/share/test/share.cypress.js:215]: Feel free to remove the comment if you are not able to reproduce the failing test.

File: src/blocks/social-profiles/test/social-profiles.cypress.js [hunk 14]
```diff
@@ -316,6 +317,7 @@ describe( 'Test CoBlocks Social Profiles Block', function() {
 	/**
 	 * Test the coblocks social profiles button size.
 	 */
+	// Same problem as share button
```
Comment (olafleur) [inline src/blocks/social-profiles/test/social-profiles.cypress.js:320]: Feel free to remove the comment if you are not able to reproduce the failing test.

File: src/components/media-filter-control/test/media-filter-control.cypress.js [hunk 15]
```diff
@@ -22,27 +22,24 @@ describe( 'Test CoBlocks Media Filter Control component', function() {
...
 			childIteration++;
 		}
 
-		i = 0;
 		childIteration = 1;
 
 		// Check the classes are applied correctly to the block
-		for ( i = 0; i < filters.length; i++ ) {
+		for ( let i = 0; i < filters.length; i++ ) {
+			// Cannot get the element without making it disappear.... strange
```
Comment (olafleur) [inline src/components/media-filter-control/test/media-filter-control.cypress.js:42]: Feel free to remove the comment if you are not able to reproduce the failing test.

File: src/extensions/lightbox-controls/test/lightbox-controls.cypress.js (hunk 6 above)
Comment (olafleur) [reply src/extensions/lightbox-controls/test/lightbox-controls.cypress.js:48]: Feel free to remove it, then. It was failing for me in local when not waiting enough time for the upload.

File: src/blocks/form/test/form.cypress.js (hunk 4 above)
Comment (olafleur) [reply src/blocks/form/test/form.cypress.js:10]: The form tests are passing for you? Here, in local, with beta3, I still have this error. If it passes on local for you and on the CI, it's probably a weirdness of my computer ;)

Comment (AnthonyLedesma) [reply src/blocks/form/test/form.cypress.js:10]: Locally for these form tests to pass it would require additonal configuration of the MailHog software. This is handled within the e2e enviornment by bash script so generally you would see differing results from From block tests locally and remote.

File: src/blocks/hero/test/hero.cypress.js (hunk 5 above)
Comment (AnthonyLedesma) [reply src/blocks/hero/test/hero.cypress.js:120]: Working for me. I have replaced the controls.

File: src/blocks/share/test/share.cypress.js (hunk 13 above)
Comment (AnthonyLedesma) [reply src/blocks/share/test/share.cypress.js:217]: Unable to reproduce. Removing.

File: src/extensions/media-text-styles/test/media-text-styles.cypress.js (hunk 7 above)
Comment (AnthonyLedesma) [reply src/extensions/media-text-styles/test/media-text-styles.cypress.js:21]: Unable to reproduce. Removing.

**Comment by AnthonyLedesma** [comment]
I've linked #1495 to be closed with this PR as a solution has been introduced.

File: src/blocks/dynamic-separator/test/dynamic-separator.cypress.js (hunk 10 above)
Comment (AnthonyLedesma) [reply src/blocks/dynamic-separator/test/dynamic-separator.cypress.js:130]: I left a few comments in places where we might encounter these errors again.

File: src/blocks/share/test/share.cypress.js (hunk 12 above)
Comment (AnthonyLedesma) [reply src/blocks/share/test/share.cypress.js]: I left a few comments in places where we might encounter these errors again.

File: .circleci/config.yml (hunk 1 above)
Comment (ehamwey) [reply .circleci/config.yml:196]: Are we ready to revert these changes @olafleur @AnthonyLedesma ?

Comment (AnthonyLedesma) [reply .circleci/config.yml:196]: Yes. I will get these reverted.

**Comment by AnthonyLedesma** [comment]
These changes are verified as backward compatible by creating CI pipelines to test against 5.8 nightly as well as 5.7.2. Test results can be seen here:
https://dashboard.cypress.io/projects/sovnn2/runs/1396/overview
772 passing e2e tests. 

The CI changes have been reverted for this PR.

**Review by ehamwey — COMMENTED** [review]
Looking good. Just to make sure since I didn't see it in my review - we already reverted the cypress config changes and fixes for the polyfills, right?

Edit: Just saw your above comment. LGTM.

**Review by ehamwey — APPROVED** [review]

**Comment by AnthonyLedesma** [comment]
I suspect that this PR should be held until additional testing can be done for Release Candidate. Added waiting label until that status changes.
